src/components/fx/hover.js

```diff
@@ -625,6 +628,46 @@ function _hover(gd, evt, subplot, noHoverEvent) {
 
     hoverData.sort(function(d1, d2) { return d1.distance - d2.distance; });
 
+    // If in compare mode, select every point at position
+    if(hoverData[0].length !== 0 &&
+      ['x', 'y'].indexOf(mode) !== -1 &&
+      hoverData[0].trace.type !== 'splom' // TODO: add support for splom
+    ) {
+        var hd = hoverData[0];
+        var cd0 = hd.cd[hd.index];
+        var isGrouped = (fullLayout.boxmode === 'group' || fullLayout.violinmode === 'group');
```

Comment:
Non-blocking:
Even for `group` which is an array of chars, one may use a const variable.